core/src/test/java/org/frankframework/parameters/NumberParameterTest.java

```diff
@@ -0,0 +1,309 @@
+package org.frankframework.parameters;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertInstanceOf;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import org.frankframework.core.ParameterException;
+import org.frankframework.core.PipeLineSession;
+import org.frankframework.stream.Message;
+import org.junit.jupiter.api.Test;
+
+public class NumberParameterTest {
+
+	@Test
+	public void testNumberWithMinLength() throws Exception {
+		NumberParameter p = new NumberParameter();
+		p.setName("number");
+		p.setValue("8");
+		p.setMinLength(10);
+		p.configure();
+
+		ParameterValueList alreadyResolvedParameters = new ParameterValueList();
+		Message message = new Message("fakeMessage");
+
+		Object result = p.getValue(alreadyResolvedParameters, message, null, false);
+		assertInstanceOf(String.class, result);
+		assertEquals("0000000008", (String) result);
+	}
+
+	@Test
+	public void testSmallNumberWithMaxLengthShouldLeftTrim() throws Exception {
+		NumberParameter p = new NumberParameter();
+		p.setName("number");
+		p.setValue("0008"); //Smaller then max-length
+		p.setMaxLength(5);
+		p.configure();
+
+		ParameterValueList alreadyResolvedParameters = new ParameterValueList();
+		Message message = new Message("fakeMessage");
+
+		Object result = p.getValue(alreadyResolvedParameters, message, null, false);
+		assertInstanceOf(Integer.class, result);
```

Comment:
Very confusing that this makes the value an Integer, but when MinLength is set it makes the value a String.
I guess that this is how the old code used to work though and that there's nothing we would want to change about this right now.

Reply:
I've added tests based on the old code. I agree that it's very strange :S